<?xml version="1.0"?>
<launch>
    <!-- action selection -->
    <arg name="output_pcd" default="true" description="Whether to output each decoded pointclouds to a PCD file"/>
    <arg name="output_rosbag" default="true" description="Whether to output the decoded pointclouds to a new RosBag"/>
    <arg name="forward_packets_to_rosbag" default="false" description="Whether to copy the input packets to the output RosBag"/>

    <!-- sensor related configuration -->
    <arg name="sensor_model"  description="Pandar64|Pandar40P|PandarXT16|PandarXT32|PandarXT32M|PandarAT128|PandarQT64|Pandar128E4X"/>
    <arg name="lidar_parameter_file" default="$(find-pkg-share nebula_ros)/config/lidar/hesai/$(var sensor_model).param.yaml" description="Path: LiDAR configuration yaml"/>

    <!-- output sample number configuration -->
    <arg name="out_num" default="0" description="The maximum number of pointclouds to output. 0 to output all pointclouds"/>
    <arg name="skip_num" default="0" description="The number of pointclouds to discard at the beginning"/>
    <arg name="bag_path" description="Path of the input RosBag"/>
    <arg name="input_topic" default="/pandar_packets" description="Name of the input packets topic, e.g. /pandar_packets"/>
    <arg name="output_topic" default="/pandar_points" description="Name of the output pointcloud topic"/>
    <arg name="out_path" description="Directory path to which the output RosBag and PCDs will be saved"/>


    <arg name="storage_id" default="sqlite3"/>
    <arg name="format" default="cdr"/>
    <arg name="only_xyz" default="true"/>


    <node pkg="nebula_examples" exec="hesai_ros_offline_extract_bag_pcd_node"
          name="hesai_cloud" output="screen">
        <param from="$(var lidar_parameter_file)" allow_substs="true" />
        <param name="output_pcd" value="$(var output_pcd)"/>
        <param name="output_rosbag" value="$(var output_rosbag)"/>
        <param name="forward_packets_to_rosbag" value="$(var forward_packets_to_rosbag)"/>
        <param name="bag_path" value="$(var bag_path)"/>
        <param name="out_path" value="$(var out_path)"/>
        <param name="input_topic" value="$(var input_topic)"/>
        <param name="output_topic" value="$(var output_topic)"/>
        <param name="storage_id" value="$(var storage_id)"/>
        <param name="format" value="$(var format)"/>
        <param name="out_num" value="$(var out_num)"/>
        <param name="skip_num" value="$(var skip_num)"/>
    </node>
</launch>
